test/unit/plugins/divclass.test.js

```diff
@@ -0,0 +1,121 @@
+const expect = require('expect');
+const parser = require('../../../src/cdoFlavoredParser');
+
+describe('divclass', () => {
+  describe('render', () => {
+    it('renders a basic divclass', () => {
+      const input = "[col-33]\n\nsimple content\n\n[/col-33]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<div class=\"col-33\"><p>simple content</p></div>\n");
+    });
+
+    it('works without content - but only if separated by FOUR newlines', () => {
+      const input = "[empty]\n\n\n\n[/empty]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<div class=\"empty\"></div>\n");
+    });
+
+    it('renders a divclass within other content', () => {
+      const input = "outside of div\n\n[divname]\n\ninside div\n\n[/divname]\n\nmore outside";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<p>outside of div</p>\n<div class=\"divname\"><p>inside div</p></div>\n<p>more outside</p>\n");
+    });
+
+    it("doesn't care about duplicate classes", () => {
+      const input = "[classname]\n\nfirst\n\n[/classname]\n\n[classname]\n\nsecond\n\n[/classname]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<div class=\"classname\"><p>first</p></div>\n<div class=\"classname\"><p>second</p></div>\n");
+    });
+
+    it('can nest divclasses', () => {
+      const input = "[outer]\n\n[inner]\n\nnested\n\n[/inner]\n\n[/outer]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<div class=\"outer\"><div class=\"inner\"><p>nested</p></div></div>\n");
+    });
```

Comment:
Test case: Can nest duplicate divclasses.
```
[foo]\n\ncontent\n\n[foo]\n\ninner content\n\n[/foo]\n\ncontent[/foo]
```

Reply:
Good call; turns out this case is actually currently broken